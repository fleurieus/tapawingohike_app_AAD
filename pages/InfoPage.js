import React from 'react';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
import { Card } from 'react-native-paper';
import { FontAwesome, FontAwesome5 } from '@expo/vector-icons';
import Header from '../components/Header';

const InfoPage = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <Card style={styles.card}>
        <Card.Content>
          <Header />
          <View style={styles.header}>
            <Text style={styles.headerText}>Hoe werkt de app?</Text>
          </View>
          <Text style={styles.text}>
            Volg de route naar het eind punt.{"\n"}
            De rode circle op het map is je doel.
          </Text>
          <View style={[styles.circle, styles.redCircle]}></View>
          <Text style={styles.text}>
            De blauwe circle geeft je huidige locatie aan.
          </Text>
          <View style={[styles.circle, styles.blueCircle]}></View>
          <Text style={styles.text}>Sommige routes hebben plaatjes en/of audio.</Text>
          <View style={styles.iconRow}>
            <FontAwesome5 name="image" size={16} color="black" style={styles.icon} />
            <Text style={styles.text}>Plaatjes laten onderdelen van de route zien - zoek deze!</Text>
          </View>
          <View style={styles.iconRow}>
            <FontAwesome5 name="volume-up" size={16} color="black" style={styles.icon} />
            <Text style={styles.text}>Luister naar de audio voor tips en info over de route.</Text>
          </View>
          <TouchableOpacity style={styles.button} onPress={() => navigation.navigate('Login')}>
            <FontAwesome name="arrow-left" size={20} color="black" />
            <Text style={styles.buttonText}>TERUG NAAR LOGIN</Text>
          </TouchableOpacity>
        </Card.Content>
      </Card>
    </View>
  );
};

export default InfoPage;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#bcebdf',
    alignItems: 'center',
    justifyContent: 'center',
  },
  card: {
    width: '80%',
    borderWidth: 1,
    borderColor: '#333',
    borderRadius: 10,
    padding: 20,
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 20,
  },
  headerText: {
    flex: 1,
    fontSize: 18,
    fontWeight: 'bold',
    marginTop: 30,
  },
  text: {
    fontSize: 16,
    marginBottom: 10,
  },
  circle: {
    width: 20,
    height: 20,
    borderRadius: 10,
    alignSelf: 'center',
    marginBottom: 20,
  },
  redCircle: {
    backgroundColor: 'red',
  },
  blueCircle: {
    backgroundColor: 'blue',
  },
  button: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 20,
    backgroundColor: '#4CAF50',
    paddingVertical: 10,
    borderRadius: 5,
  },
  buttonText: {
    marginLeft: 10,
    fontSize: 16,
    color: 'white',
  },
  iconRow: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  icon: {
    marginRight: 10,
  },
});
