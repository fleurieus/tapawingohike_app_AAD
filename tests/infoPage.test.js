import React from 'react';
import renderer from 'react-test-renderer';
import InfoPage from '../pages/InfoPage'; // Adjust the import path as per your project structure

// Mock navigation prop
const mockNavigation = {
  navigate: jest.fn(),
};

describe('<InfoPage />', () => {
  it('renders correctly', () => {
    const component = renderer.create(<InfoPage navigation={mockNavigation} />);
    const tree = component.toJSON();

    // Assert if the component renders correctly
    expect(tree).toMatchSnapshot();

    // Check the number of children at the top level
    const topLevelChildren = tree.children[0].children;
    expect(topLevelChildren.length).toBe(1); // Adjust this based on your component structure
  });
});
