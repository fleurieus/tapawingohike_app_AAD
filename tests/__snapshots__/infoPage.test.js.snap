// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoPage /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#bcebdf",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(247, 243, 249, 1)",
        "borderRadius": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
        "width": "90%",
      }
    }
    testID="card-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(247, 243, 249, 1)",
          "borderColor": "#333",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 1,
          "flex": undefined,
          "padding": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
      testID="card-container"
    >
      <View
        style={
          [
            {
              "flexShrink": 1,
            },
            undefined,
          ]
        }
        testID="card"
      >
        <View
          borderRadiusStyles={
            {
              "borderRadius": 10,
            }
          }
          style={
            [
              {
                "paddingHorizontal": 16,
              },
              {
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#f0f0f0",
                "borderBottomWidth": 4,
                "borderColor": "#000000",
                "borderTopWidth": 4,
                "flexDirection": "row",
                "paddingHorizontal": 20,
                "paddingVertical": 10,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 60,
                  "marginRight": 10,
                  "paddingLeft": 10,
                  "width": 60,
                }
              }
              testID="logo-image"
            />
            <View
              style={
                {
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 28,
                    "fontWeight": "bold",
                  }
                }
              >
                Tapawingo
              </Text>
              <Text
                style={
                  {
                    "fontSize": 20,
                    "marginLeft": 20,
                  }
                }
              >
                Hike app
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 20,
              }
            }
          >
            <Text
              style={
                {
                  "flex": 1,
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginTop": 30,
                }
              }
            >
              Hoe werkt de app?
            </Text>
          </View>
          <Text
            style={
              {
                "fontSize": 16,
                "marginBottom": 10,
              }
            }
          >
            Volg de route naar het eind punt.
            

            De rode marker op het map is je doel.
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "marginBottom": 20,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
          <Text
            style={
              {
                "fontSize": 16,
                "marginBottom": 10,
              }
            }
          >
            De blauwe circle geeft je huidige locatie aan.
          </Text>
          <View
            style={
              [
                {
                  "alignSelf": "center",
                  "borderRadius": 10,
                  "height": 20,
                  "marginBottom": 20,
                  "width": 20,
                },
                {
                  "backgroundColor": "blue",
                },
              ]
            }
          />
          <Text
            style={
              {
                "fontSize": 16,
                "marginBottom": 10,
              }
            }
          >
            Sommige routes hebben plaatjes en/of audio.
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 10,
              }
            }
          >
            <Text />
            <Text
              style={
                {
                  "fontSize": 16,
                  "marginBottom": 10,
                }
              }
            >
              Plaatjes laten onderdelen van de route zien - zoek deze!
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 10,
              }
            }
          >
            <Text />
            <Text
              style={
                {
                  "fontSize": 16,
                  "marginBottom": 10,
                }
              }
            >
              Luister naar de audio voor tips en info over de route.
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#4CAF50",
                "borderRadius": 5,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 20,
                "opacity": 1,
                "paddingVertical": 10,
              }
            }
          >
            <Text />
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 16,
                  "marginLeft": 10,
                }
              }
            >
              TERUG NAAR LOGIN
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
